x-pack/plugins/apm/public/locator/helpers.ts

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import * as t from 'io-ts';
+import { isRight } from 'fp-ts/Either';
+import { PathReporter } from 'io-ts/lib/PathReporter';
+import { environmentRt } from '../../common/environment_rt';
+import { apmRouter } from '../components/routing/apm_route_config';
+import type { TimePickerTimeDefaults } from '../components/shared/date_picker/typings';
+
+export const APMLocatorPayloadValidator = t.intersection([
+  t.type({
+    serviceName: t.string,
+  }),
+  t.partial({
+    serviceOverviewTab: t.keyof({
+      traces: null,
+      metrics: null,
+      logs: null,
+    }),
+  }),
+  t.type({
+    query: environmentRt,
+  }),
+]);
+
+export type APMLocatorPayload = t.TypeOf<typeof APMLocatorPayloadValidator>;
+
+export function getPathForServiceDetail(
+  payload: APMLocatorPayload,
+  {
+    from,
+    to,
+    isComparisonEnabledByDefault,
+  }: TimePickerTimeDefaults & { isComparisonEnabledByDefault: boolean }
+) {
+  const decodedPayload = APMLocatorPayloadValidator.decode(payload);
+  if (!isRight(decodedPayload)) {
+    throw new Error(PathReporter.report(decodedPayload).join('\n'));
+  }
+
+  const mapObj = {
+    logs: '/services/{serviceName}/logs',
```

Comment:
Are we supposed to link to logs here? What about errors?

Reply:
Based on [the ecs data stream doc](https://www.elastic.co/guide/en/ecs/current/ecs-data_stream.html#ecs-data_stream), the stream type can only be one of these values `metrics`, `logs`, or `traces`, let me know if I misunderstood it?

Reply:
Yes, I think errors are stored in logs though. Not sure. Can you see where you get hits for processor.event:error? You can use Discover for this. If they're in the logs data stream, can you clarify with product where we should link to in that case?

Reply:
I double checked the logs documents and found that you are right, errors are stored in logs with the `processor.event` containing the value `error`, changed in https://github.com/elastic/kibana/pull/130251/commits/088431abdb19e3ef3011a53565ec495bc70c1585